pkg/services/ngalert/migration/models/alertmanager.go

```diff
@@ -0,0 +1,83 @@
+package models
+
+import (
+	"github.com/prometheus/alertmanager/config"
+	"github.com/prometheus/alertmanager/pkg/labels"
+	"github.com/prometheus/common/model"
+
+	apiModels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	ngmodels "github.com/grafana/grafana/pkg/services/ngalert/models"
+)
+
+// UseLegacyChannelsLabel is a private label added to a rule's labels to enable routing to the nested route created
+// during migration.
+const UseLegacyChannelsLabel = "__use_legacy_channels__"
+
+// Alertmanager is a helper struct for creating migrated alertmanager configs.
+type Alertmanager struct {
+	Config      *apiModels.PostableUserConfig
+	legacyRoute *apiModels.Route
+}
```

Comment:
Why not use apimodels.PostableApiAlertingConfig instead and just implement some utility functions?

Reply:
No reason other than I found this made the code simpler. It makes a bit more sense in a future PR where `NewAlertmanager` is replaced by `FromPostableUserConfig`, but overall it just hides implementation details around where the routes are stored (and later how to clean them up).